pkg/storage/bucket/http/config.go

```diff
@@ -2,20 +2,38 @@ package http
 
 import (
 	"flag"
+	"net/http"
 	"time"
 )
 
+// NOTE some of the fields are hidden in the documentation due to this struct
+// being by the old Swift storage backend. The hidden fields can be unhidden
+// when we deprecate the old clients.
+
 // Config stores the http.Client configuration for the storage clients.
 type Config struct {
-	IdleConnTimeout       time.Duration `yaml:"idle_conn_timeout"`
-	ResponseHeaderTimeout time.Duration `yaml:"response_header_timeout"`
-	InsecureSkipVerify    bool          `yaml:"insecure_skip_verify"`
-	TLSHandshakeTimeout   time.Duration `yaml:"tls_handshake_timeout"`
-	ExpectContinueTimeout time.Duration `yaml:"expect_continue_timeout"`
-	MaxIdleConns          int           `yaml:"max_idle_connections"`
-	MaxIdleConnsPerHost   int           `yaml:"max_idle_connections_per_host"`
-	MaxConnsPerHost       int           `yaml:"max_connections_per_host"`
-	CAFile                string        `yaml:"ca_file"`
+	IdleConnTimeout       time.Duration `yaml:"idle_conn_timeout" doc:"hidden"`
```

Comment:
can we also this common `http.Config` in [s3 config](https://github.com/grafana/loki/blob/f2da621f7255c678774fb0e99c0b7fb68f6a6a69/pkg/storage/bucket/s3/config.go#L59) to avoid duplicate code?

Reply:
Yes, I had planned it as a follow up but I'm happy to make it here